tiled/src/main/kotlin/ktx/tiled/tiledMaps.kt

```diff
@@ -153,3 +154,18 @@ inline fun TiledMap.forEachMapObject(layerName: String, action: (MapObject) -> U
     action(it)
   }
 }
+
+/**
+ * Extension method to run a [lambda] on a specific type of [layers][MapLayer] of the [TiledMap]. The lambda
+ * takes the matching [MapLayer] as a parameter.
+ *
+ * The class matching is an exact match meaning that a given subclass like [TiledMapTileLayer] will not match
+ * for the argument [MapLayer].
+ */
+inline fun <reified T : MapLayer> TiledMap.forEachLayer(lambda: (T) -> Unit) {
+  this.layers.forEach {
+    if (it::class == T::class) {
```

Comment:
Is there a specific reason why you're using an exact class match? Is it like in the example with `MapLayer`, where it is used both for extension and as a concrete class for specific layers?

Reply:
Tiled has four specific layer types currently:
- Tiled
- Object
- Image
- Group

For whatever reason LibGDX only implemented three of them: `MapGroupLayer`, `TiledMapImageLayer` and `TiledMapTileLayer`. And they have a generic `MapLayer` class.

The way I needed this extension was that I wanted to iterate over Object layers, get their objects and create entities out of it. Since the object layer is just a generic `MapLayer` I would get all layers again if I use an instance check.

Also, when you use `forEachLayer` you most likely want to do that for one specific layer type.

![image](https://user-images.githubusercontent.com/93260/110015024-bc374c00-7d23-11eb-8410-b65d56171fb9.png)